### PR TITLE
Flutter scanner improvements

### PR DESCRIPTION
**Pull request by godrei**
### Context

First builds for Flutter projects are regularly failing, because the preinstalled Flutter SDK on stacks is not satisfy the given project's Flutter or Dart SDK requirements.

This PR updates the Flutter scanner, to read the given project's Flutter and Dart SDK requirements and configure the Flutter Installer Step based on that.

Related ticket: https://bitrise.atlassian.net/browse/BIVS-1852

### Changes

**[go-flutter package](https://github.com/bitrise-io/go-flutter/pull/2)**

Flutter project parsing logic moved to the go-flutter project in order to share common logic with Steps. Currently reading the Flutter and Dart SDK requirements are shared with the Flutter Installer step.

The Flutter scanner utilizes this package to populate an internal project struct with the necessary information for config generation.

**Flutter Installer Step**

The Flutter Installer step is configured based on the project's SDK requirements.

**Generated configs**

- For library projects, no deploy workflow will be generated as flutter build fails for such a project
- If a project doesn't have tests the primary workflow will contain the flutter-analyse step

**Manual configs**

- The manual config was not utilizing the provided project location, this is fixed
- The manual config contained configs for projects with and without tests, but only configs with tests were reachable, so the configs without test were removed

**Improvements**

- Xcode project parsing is removed because the information gathered was unused.
- `getBuildablePlatform` (as `platform`) and generated config name calculation (as `configName`) are assigned to the `project` struct
- Flutter scanner was generating all the possible config variations for the generated automatic configs, regardless of the project properties

### CODE REVIEW

File: scanners/flutter/flutter.go [hunk 1]
```diff
@@ -244,108 +159,129 @@ func (Scanner) ExcludedScannerNames() []string {
...
 
 // DefaultOptions ...
-func (Scanner) DefaultOptions() models.OptionNode {
+func (scanner *Scanner) DefaultOptions() models.OptionNode {
 	flutterProjectLocationOption := models.NewOption(projectLocationInputTitle, projectLocationInputSummary, projectLocationInputEnvKey, models.TypeUserInput)
 
-	cfg := configName + "-test"
-
 	flutterPlatformOption := models.NewOption(platformInputTitle, platformInputSummary, "", models.TypeSelector)
-	flutterProjectLocationOption.AddOption("", flutterPlatformOption)
```
Comment (godrei) [inline scanners/flutter/flutter.go:281]: empty string for user_input type could be a descriptive constant

File: _tests/integration/manual_config_test.go [hunk 2]
```diff
@@ -371,18 +321,18 @@ var customConfigResultYML = fmt.Sprintf(`options:
     env_key: BITRISE_FLUTTER_PROJECT_LOCATION
     type: user_input
     value_map:
-      "":
+      $BITRISE_FLUTTER_PROJECT_LOCATION:
```
[the remark this answers is not in the document]
Comment (godrei) [reply _tests/integration/manual_config_test.go]: updated in: https://github.com/bitrise-io/bitrise-init/pull/258/commits/ac32cb76925b36d332dda40110eef71fc05bd2d4

File: scanners/flutter/flutter.go (hunk 1 above)
Comment (godrei) [reply scanners/flutter/flutter.go:281]: Updated in: https://github.com/bitrise-io/bitrise-init/pull/258/commits/78f2ffa6aa3ca8cc84b3323761c6ee621299e40f

File: scanners/flutter/flutter.go [hunk 3]
```diff
@@ -244,108 +159,129 @@ func (Scanner) ExcludedScannerNames() []string {
...
+
+	// restore cache is after flutter-installer, to prevent removal of pub system cache
+	configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.RestoreDartCache())
+
+	if proj.hasTest {
+		configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.FlutterTestStepListItem(
+			envmanModels.EnvironmentItemModel{projectLocationInputKey: "$" + projectLocationInputEnvKey},
+		))
+	} else {
+		configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.FlutterAnalyzeStepListItem(
```
Comment (ofalvai) [inline scanners/flutter/flutter.go:269]: Isn't it useful to include the analyze step in every workflow, even if it has tests?

Comment (godrei) [reply scanners/flutter/flutter.go:269]: It seems earlier we intentionally removed the analyze step from primary workflows: https://github.com/bitrise-io/bitrise-init/pull/223

**Pull request merged**